src/index.js

```diff
@@ -1,452 +1,397 @@
+   * @param {string} hookURL
+   * @param {string} apiKey
+   * @returns {Promise<Response | never>}
+   */
+  getBotName = (hookURL = this.configuration.hookURL, apiKey = this.configuration.apiKey) =>
+    fetch(`${hookURL}?key=${apiKey}`, {
+      method: 'GET',
+    }).then(response => {
+      if (!response.ok) {
+        throw response;
+      }
+      return response.json();
+    });
 
-      if (template.template_type === TemplateTypes.FB_MESSENGER_BUTTON) {
-        textFromBot = template.payload.text;
+  /**
+   * - Initialises the widget
+   * - Resolves promise on connection success with botName and history
+   * - Rejects promise if it fails
+   * @returns {Promise<object>}
+   * where object = {botName: string, history: Array[{content: string, made_by: string}]}
+   */
+  init = () => {
+    this.inSession = true;
+
+    return new Promise((resolve, reject) => {
+      // store promise to resolve on establish session success
+      this.promises.init = { resolve, reject };
+      this._setUpSocket();
+    });
+  };
 
-        store.onMessage(textFromBot);
-      } else if (template.template_type === TemplateTypes.FB_MESSENGER_QUICK_REPLY) {
-        textFromBot = template.text;
+  /**
+   * Set callback for incoming messages
+   * @param {function} callback callback for incoming messages
+   */
+  onMessage = callback => {
+    this.onMessageCallback = callback;
+  };
 
-        store.onMessage(textFromBot);
+  /**
+   * Sends a message if connected, else queues it
+   * @param {string} message
+   */
+  send = message => {
+    if (this.connected) {
+      postMessage({
+        url: this.configuration.hookURL,
+        apiKey: this.configuration.apiKey,
+        input: message,
+        userId: this.userId,
+        payload: this.payload,
+      });
+
+      this.chatHistoryManager.push({ made_by: 'user', content: message });
+    } else {
+      if (this.messageQueue.length === 1 && this.messageQueue[0] === CONVERSATION_STARTER) {
+        this.messageQueue = [];
       }
+
+      this.messageQueue.push(message);
     }
+  };
 
-    textFromBot = data.emission;
+  /**
+   * - To get the bot to say the intro message (if exists)
+   * without having the user to send a message first
+   * - Must be called after init
+   */
+  initiateConversation = () => {
+    this.send(CONVERSATION_STARTER);
+  };
 
-    store.onMessage(textFromBot);
-    chatHistory.push({ made_by: 'chatbot', content: textFromBot });
-  }
-};
+  /**
+   * - Closes socket connection for good
+   * - Clears callback references
+   */
+  endSession = () => {
+    // toggle session flag first so socket doesn't reconnect
+    this.inSession = false;
+
+    if (this.socket) this.socket.close();
+
+    this.onMessage = f => f;
+    this.promises.getUserId = null;
+    this.messageQueue = [];
+    this.onConnectionChangeCallback = f => f;
+    this._clearInitPromise();
+  };
 
-const setUpSocket = () => {
-  if (!store.notificationsChannelURL) {
```

Comment:
Is this maintained still? Or do we not need this anymore?

https://github.com/twyla-ai/widget-core/commit/1922d118172d29e2f54e6e92c8e922e5b42e7c36

Reply:
oops I missed this too. So this would be caught while parsing the hook URL right? So we don't need this.